### ISSUE TITLE
Forces CSD despite use of suitably-patched gtk3-nocsd

### ISSUE THREAD

**Issue by dsalt** [wwmm/easyeffects#3453]
CSD makes this application _inconsistent_ here (Xfce4 desktop), both in appearance and in decoration functionality; (almost) everything else which I use uses server-side decorations natively, can be forced to without causing problems or provides an option.

This bug (or misfeature, if you prefer – it's certainly unwanted behaviour here) appears to be caused by use of adwaita.

**Comment by wwmm** [comment]
EasyEffects uses gtk4 and not gtk3... In any case I am not going to start a discussion about client side vs server decoration for several reasons:

- Compatibility with desktops that do not use CSD is something that should be discussed with the library (gtk) devs. Not here. I do not know how familiar you are with graphical toolkits. But this isn't something that should be fixed in the clients. And expecting people to support patches that are not coming from the libraries developers is quite unreasonable...
- In most cases CSD vs SSD is a matter of personal taste
- For different reasons that having nothing to do with this endless and pointless CSD vs SSD debate I started a Qt port. There is not point in doing these kind of changes to the gtk branch. Assuming it is is even possible to force gtk4 to not use CSD without removing CSD from the people that like CSD...

**Comment by dsalt** [comment]
gtk3-nocsd, at least as found in Ubuntu, has patches for handling GTK4. (I'm using a local build of that.)

As it seems to be adwaita which is the problem here, I think that GNOME devs would be more appropriate. (My understanding is that GTK allows for either, but GNOME enforces CSD.)

Regardless, the choice is there not to use that and to use plain GTK or something else (practicality notwithstanding).

I hadn't realised that there was a Qt branch – I've just had a quick look and very quickly noticed that the build-deps in debian/control need to be corrected (`meson (>= 3.28)`, removal of GTK/GNOME deps, addition of Qt6 deps). (Also, looks like some deps aren't in Devuan stable or stable-backports. Oh well.)

**Comment by wwmm** [comment]
>I hadn't realised that there was a Qt branch

It is far from being in an usable state. So I did not pin an issue to discuss its development yet.

> I've just had a quick look and very quickly noticed that the build-deps in debian/control need to be corrected (cmake (>= 3.28), removal of GTK/GNOME deps, addition of Qt6 deps).

I never took part in the Debian scripts development and maintenance because I can't test it. I use Arch Linux. Those scripts are here only because it was convenient for Debian package maintainers.

In any case I think it is too soon to put effort on them.  It is better to wait for the qt branch to be in a state where it can be properly tested.

**Comment by wwmm** [comment]
>Regardless, the choice is there not to use that and to use plain GTK or something else (practicality notwithstanding).

There is a choice only on paper. I tried to not use libadwaita in the past. The issue is that pure gtk does not offer some of the useful widgets that are provided only by libadwaita. And trying to implementing them in pure gtk is a pain. I attempted this and it was not a nice experience. Sure it can be done. But the amount of time and effort required for this is not small. In the end just using libadwaita is more practical.
